test/index.ts

```diff
@@ -832,6 +833,17 @@ function testNonEmptyObject() {
   type TestB = Assert<IsExact<NonEmptyObject<EmptyObject>, never>>;
 }
 
+function testNonEmptyArray() {
+  type Cases<T> = [
+    Assert<IsExact<NonEmptyArray<T>, [T, ...T[]]>>,
+    AssertFalse<IsExact<NonEmptyArray<T>, []>>,
+
+    AssertFalse<Assignable<NonEmptyArray<T>, []>>,
+    Assert<Assignable<NonEmptyArray<T>, [T]>>,
+    Assert<Assignable<NonEmptyArray<T>, [T, ...T[]]>>,
+  ];
+}
```

Comment:
Thank you for the change and all in all, smart idea with generic type in `Cases<T>`!